### PR TITLE
Support for DarkTheme in Switch

### PR DESCRIPTION
**Pull request by pooja-ps-trimble**

### CODE REVIEW

File: Trimble.Modus.Components/Controls/TMSwitch/TMSwitch.xaml.cs [hunk 1]
```diff
@@ -169,7 +221,8 @@ private static void OnSwitchSelected(TMSwitch tMSwitch)
                 { 0, 1, new Animation(v => tMSwitch.circle.TranslationX = v, tMSwitch.circleMargin,tMSwitch.border.WidthRequest - tMSwitch.circle.WidthRequest - tMSwitch.circleMargin) }
             },
             length: 250, easing: Easing.CubicIn);
-        tMSwitch.border.Color = ResourcesDictionary.ColorsDictionary(ColorsConstants.BlueLight);
+        VisualStateManager.GoToState(tMSwitch, "Selected");
+        UpdateSwitchStyle(tMSwitch);
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMSwitch/TMSwitch.xaml.cs:225]: I don't think you have to update the style everytime the state changes. Can you remove this and check?

File: Trimble.Modus.Components/Controls/TMSwitch/TMSwitchStyles.xaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+                    xmlns:local="clr-namespace:Trimble.Modus.Components;assembly=Trimble.Modus.Components"
+                    x:Class="Trimble.Modus.Components.TMSwitchStyles">
+
+  <Style TargetType="local:TMSwitch" x:Key="SwitchStyle">
+    <Setter Property="TextColor" Value="{DynamicResource SwitchText}"/>
+    <Setter Property="VisualStateManager.VisualStateGroups">
+      <VisualStateGroupList>
+        <VisualStateGroup x:Name="CommonStates">
+          <VisualState x:Name="Selected">
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMSwitch/TMSwitchStyles.xaml:12]: Switch should have a 'On' and 'Off' state. Default MAUI switch has those. Can you follow the same here?

https://learn.microsoft.com/en-us/xamarin/xamarin-forms/user-interface/switch#switch-visual-states

File: Trimble.Modus.Components/Styles/DarkTheme.xaml [hunk 3]
```diff
@@ -73,6 +73,12 @@
   <Color x:Key="BadgeAlternateTextColor">#252A2E</Color>
   <Color x:Key="BadgeTextColor">#FFFFFF</Color>
 
+  <!--Switch-->
+  <Color x:Key="SwitchSelected">#019AEB</Color>
```
Comment (rthanga1) [inline Trimble.Modus.Components/Styles/DarkTheme.xaml:77]: `SwitchOn BackgroundColor` , `SwitchOffBackgroundColor` will make more sense.

File: Trimble.Modus.Components/Styles/DarkTheme.xaml [hunk 4]
```diff
@@ -73,6 +73,12 @@
...
+  <Color x:Key="SwitchUnselected">#90939F</Color>
+  <Color x:Key="SwitchText">#FFFFFF</Color>
```
Comment (rthanga1) [inline Trimble.Modus.Components/Styles/DarkTheme.xaml:79]: `SwitchTextColor`

**Review by rthanga1 — CHANGES_REQUESTED** [review]

File: Trimble.Modus.Components/Controls/TMSwitch/TMSwitch.xaml.cs (hunk 1 above)
Comment (pooja-ps-trimble) [reply Trimble.Modus.Components/Controls/TMSwitch/TMSwitch.xaml.cs]: Yes. It wasn't necessary. I removed it.Yes. It wasn't necessary. I removed it.

File: Trimble.Modus.Components/Styles/DarkTheme.xaml (hunk 3 above)
Comment (pooja-ps-trimble) [reply Trimble.Modus.Components/Styles/DarkTheme.xaml]: Sure

**Review by rthanga1 — APPROVED** [review]

**Review by rthanga1 — APPROVED** [review]

**Pull request merged**